beacon-chain/core/helpers/validators.go

```diff
@@ -179,11 +179,60 @@ func BeaconProposerIndex(state *stateTrie.BeaconState) (uint64, error) {
 		return 0, errors.Wrap(err, "could not update committee cache")
 	}
 
-	return ComputeProposerIndex(state.Validators(), indices, seedWithSlotHash)
+	return ComputeProposerIndex(state, indices, seedWithSlotHash)
 }
 
 // ComputeProposerIndex returns the index sampled by effective balance, which is used to calculate proposer.
```

Comment:
Somewhere in the comment, can you please mention:
1.) This is faster than `ComputeProposerIndexWithValidators`
2.) It uses read only validator set from beacon state

Reply:
Alright done @terencechain